### PR TITLE
Feature: set language in user data

### PR DESCRIPTION
**Pull request by josadcha**

### CODE REVIEW

File: src/containers/Sidebar/index.tsx [hunk 1]
```diff
@@ -193,6 +198,16 @@ class SidebarContainer extends React.Component<Props, State> {
 
 
     private handleChangeLanguage = (language: string) => {
+        const { user, isLoggedIn } = this.props;
+        if (isLoggedIn) {
+            const payload = {
+                ...user,
+                data: JSON.stringify({ language }),
```
Comment (oyershov) [inline src/containers/Sidebar/index.tsx:205]: Don`t override data like this. Add new one with spread.
```
data: JSON.stringify({
    ...JSON.parse(user.data),
    language,
)},
```

File: src/modules/public/i18n/reducer.ts [hunk 2]
```diff
@@ -6,7 +7,7 @@ export interface LanguageState {
 }
 
 const defaultLanguage = {
-    code: 'en',
+    code: languages[0],
```
Comment (oyershov) [inline src/modules/public/i18n/reducer.ts:10]:
```suggestion
    code: languages[0] || 'en',
```

File: src/modules/user/profile/reducer.ts [hunk 3]
```diff
@@ -173,6 +177,29 @@ export const userReducer = (state: ProfileState['userData'], action: ProfileActi
...
+            return {
+                ...state,
+                success: false,
+            };
+        case PROFILE_CHANGE_USER_DATA:
+            return {
+                ...state,
+                user: {
+                    ...state.user,
+                    data: action.payload.user.data,
```
Comment (oyershov) [inline src/modules/user/profile/reducer.ts:190]: You should handle case when there are:
1. ```action.payload``` is null
2. ```action.payload.user``` is null

File: src/modules/user/profile/reducer.ts [hunk 4]
```diff
@@ -173,6 +177,29 @@ export const userReducer = (state: ProfileState['userData'], action: ProfileActi
...
+                },
+                success: true,
+            };
+        case PROFILE_CHANGE_USER_ERROR:
+            return {
+                ...state,
+                user: {
+                    ...state.user,
+                    data: state.user.data,
+                },
```
Comment (oyershov) [inline src/modules/user/profile/reducer.ts:200]: No need to override user.
```
return {
    ...state,
    success: false,
}
```

File: src/modules/public/i18n/reducer.ts (hunk 2 above)
Comment (josadcha) [reply src/modules/public/i18n/reducer.ts:10]: export const languages = Cryptobase.config.languages && Cryptobase.config.languages.length > 0 ? Cryptobase.config.languages : ['en'];

it is already elaborated in config.ts

File: src/modules/user/profile/reducer.ts [hunk 5]
```diff
@@ -187,17 +187,13 @@ export const userReducer = (state: ProfileState['userData'], action: ProfileActi
                 ...state,
                 user: {
                     ...state.user,
-                    data: action.payload.user.data,
+                    data: action.payload && action.payload.user && action.payload.user.data || state.user.data,
```
Comment (oyershov) [inline src/modules/user/profile/reducer.ts:190]: Please, don\`t use mixed operators.
Isn`t there linter error ```no-mixed-operators```?

File: src/modules/user/profile/sagas/changeUserDataSaga.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+// tslint:disable-next-line
```
Comment (akhlopiachyi) [inline src/modules/user/profile/sagas/changeUserDataSaga.ts:1]: remove or clarify the reason

**Pull request merged**